### DISCUSSION TITLE
About Issue 2705

### DISCUSSION THREAD

**Discussion by OlivierDoriath** [86Box/86Box#2716] [Q&A]
About the issue I posted here: https://github.com/86Box/86Box/issues/2705

After a little bit of digging, I found that replacing 

`define BLOCK_GPF_OFFSET (BLOCK_EXIT_OFFSET - 15)
`
with

`define BLOCK_GPF_OFFSET (BLOCK_EXIT_OFFSET - 11)
`
in src\codegen\codegen_x86-64.h and src\codegen\codegen_x86.h at line 14 seems to be enough to fix the issue. The intent of the number 15 isn't clear though, and the reasoning behind

```
if (block_pos > BLOCK_GPF_OFFSET)
        fatal("Over limit!\n");
```

in codegen_x86-64.c and codegen_x86.c isn't either, and I could be wrong. Would someone be willing to discuss this?

**Comment by OBattler** [comment]
The number is how many bytes behind the block exit handler is the GPF handler, which is also why the fatal occurs - the GPF handler is being overwritten. But I need to look at the number - it may indeed be wrong.

**Reply by OBattler** [reply]
I just checked - the numbers are in fact correct.

**Reply by OBattler** [reply]
OK, I need the VM so I can reproduce the bug and find a way to fix it.

**Reply by OlivierDoriath** [reply]
Dropbox link sent on Discord 👍

**Comment by dhrdlicka** [comment]
For the record, there is no need to create a separate discussion for stuff that is related to an existing issue/discussion. In the future, just post to the original one.

**Reply by OlivierDoriath** [reply]
Sorry! I wasn't sure if discussions are encouraged on bug reports. Thanks for the tip :)
